src/Components/Blazor/Validation/src/BlazorDataAnnotationsValidator.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    public class BlazorDataAnnotationsValidator : ComponentBase
```

Comment:
Is this meant to appear in the user markup?

I would try to pick a better name that makes it clear what it does, even if its a bit longer than the current identifier. Some  options are:
* GraphDataAnnotationsValidator
* ObjectGraphDataAnnotationsValidator
* RecursiveDataAnnotationsValidator

In essence, something that clearly states what this does.